### PR TITLE
Multiple model wrapping server. For Multilingual Demos

### PR DESCRIPTION
**Pull request by shyamupa**

### CODE REVIEW

File: ccg_nlpy/pipeline_base.py [hunk 1]
```diff
@@ -83,5 +81,5 @@ def add_additional_views_to_TA(self, textannotation, views):
         @return: raw text of the response from server -- jsonStr
         """
         logger.error("This function should be overrided.")
-        #raise NotImplementedError()
+        # raise NotImplementedError()
```
Comment (danyaljj) [inline ccg_nlpy/pipeline_base.py]: Argument types: nice work!

File: ccg_nlpy/server/abstract_model.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+from typing import List
+
+from ccg_nlpy.core.text_annotation import TextAnnotation
+
+
+class AbstractModel:
```
Comment (danyaljj) [inline ccg_nlpy/server/abstract_model.py]: Just keep things compatible with the Java side, how about we name it sth like `AbstractView`?

File: ccg_nlpy/server/abstract_model.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    def load_params(self) -> None:
+        """
+        Load the relevant model parameters.
+        :return: None
+        """
+        raise NotImplementedError
+
+    def get_provided_view(self) -> str:
```
Comment (danyaljj) [inline ccg_nlpy/server/abstract_model.py]: Java compatibility: `get_view_name`

File: ccg_nlpy/server/abstract_model.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        """
+        Return the name of the view that will be provided by the model.
+        :return: viewName
+        """
+        raise NotImplementedError
+
+    def inference_on_ta(self, docta: TextAnnotation) -> TextAnnotation:
```
Comment (danyaljj) [inline ccg_nlpy/server/abstract_model.py]: CogCompNLP (java) compatibility: `add_name`

File: ccg_nlpy/server/abstract_model.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        """
+        Takes a text annotation and adds the view provided by this model to it.
+        :return: viewName
```
Comment (danyaljj) [inline ccg_nlpy/server/abstract_model.py]: returns a TextAnnotation.

File: ccg_nlpy/server/abstract_model.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+        """
+        raise NotImplementedError
+
+    def get_required_views(self) -> List[str]:
+        """
+        Return the list of viewnames required by the model.
+        :return: viewName
```
Comment (danyaljj) [inline ccg_nlpy/server/abstract_model.py]: returns a list of strings.

File: ccg_nlpy/server/abstract_model.py (hunk 2 above)
Comment (shyamupa) [reply ccg_nlpy/server/abstract_model.py]: What do we call a model that works inside a server? I can name this `Annotator`

Comment (danyaljj) [reply ccg_nlpy/server/abstract_model.py]: Makes sense.

**Comment by danyaljj** [comment]
Side issue: Semaphore is failing a couple of tests.

**Comment by danyaljj** [comment]
I think you have to update your branch with master.

**Comment by shyamupa** [comment]
Just did. Try again maybe?

**Pull request merged**